packages/react/src/components/Slider/Slider.tsx

```diff
@@ -0,0 +1,276 @@
+    _.invoke(this.props, 'onMouseDown', e, this.props)
+  }
+
+  renderComponent({
+    ElementType,
+    classes,
+    accessibility,
+    variables,
+    styles,
+    unhandledProps,
+  }: RenderResultConfig<SliderProps>) {
+    const { input, inputRef, step } = this.props
+    const [htmlInputProps, restProps] = partitionHTMLProps(unhandledProps)
+    const type = 'range'
+
+    const { min, max, value, valueAsPercentage } = processInputValues({
+      min: this.props.min,
+      max: this.props.max,
+      value: this.state.value || '',
+    })
+
+    // we need 2 wrappers around the slider rail, track, input and thumb slots to achieve correct component sizes
+    return (
+      <ElementType
+        className={classes.root}
+        {...accessibility.attributes.root}
+        {...restProps}
+        {...applyAccessibilityKeyHandlers(accessibility.keyHandlers.root, unhandledProps)}
+      >
+        <div className={cx(Slider.slotClassNames.inputWrapper, classes.inputWrapper)}>
+          <span className={cx(Slider.slotClassNames.rail, classes.rail)} />
+          <span
+            className={cx(Slider.slotClassNames.track, classes.track)}
+            style={{ width: valueAsPercentage }}
+          />
+          <Ref
+            innerRef={(inputElement: HTMLElement) => {
+              handleRef(this.inputRef, inputElement)
+              handleRef(inputRef, inputElement)
+            }}
+          >
+            {Box.create(input || type, {
+              defaultProps: {
+                ...htmlInputProps,
+                ...accessibility.attributes.input,
+                className: Slider.slotClassNames.input,
+                as: 'input',
+                min,
+                max,
+                step,
+                type,
+                value,
+                onChange: this.handleChange,
+                onFocus: this.handleFocus,
+                onMouseDown: this.handleMouseDown,
+                styles: styles.input,
+                ...applyAccessibilityKeyHandlers(accessibility.keyHandlers.input, htmlInputProps),
+              },
+            })}
+          </Ref>
+          {/* the thumb slot needs to appear after the input slot */}
+          <span
+            className={cx(Slider.slotClassNames.thumb, classes.thumb)}
+            style={{ left: valueAsPercentage }}
+          />
+        </div>
+      </ElementType>
+    )
+  }
+}
+
+Slider.slotClassNames = {
+  input: `${Slider.className}__input`,
+  inputWrapper: `${Slider.className}__input-wrapper`,
+  rail: `${Slider.className}__rail`,
+  thumb: `${Slider.className}__thumb`,
+  track: `${Slider.className}__track`,
+}
+
+/**
+ * A slider is an input that allows the user to choose a value from within a specific range of values.
+ * @accessibility
+ * Implements [ARIA Slider](https://www.w3.org/TR/wai-aria-practices-1.1/#slider) design pattern.
+ */
+export default withSafeTypeForAs<typeof Slider, SliderProps, 'input'>(Slider)
```

Comment:
when doing this, I cannot access `Slider.defaultProps` outside of this file, @kuzhelov any idea why?

![Screenshot 2019-07-12 at 13 38 58](https://user-images.githubusercontent.com/5442794/61125564-6ce73c80-a4aa-11e9-9d26-93832cc68a9f.png)